crates/uv-client/src/base_client.rs

```diff
@@ -417,6 +449,16 @@ impl BaseClient {
         }
     }
 
+    /// Executes a request. If a 302 response is encountered, tries the request again through
+    /// the entire middleware pipeline with the redirect URL from the 302.
+    pub async fn execute_with_redirect_handling(
```

Comment:
(*) How did you decide which callers use `execute_with_redirect_handling` and which use `for_host`?